.github/workflows/build.yml

```diff
@@ -48,20 +51,36 @@ jobs:
     - name: Bootstrap
       uses: ./.github/bootstrap_platform
 
-    - name: Configure CMake build
+     # We don't want to publish the test build, it gets too big.
+    - name: Build and Install (No tests)
       run: >
         ${{ matrix.config.preamble }}
 
         cmake -S . -B build -G Ninja
         --install-prefix ${{ github.workspace }}/dist
         --toolchain ${{ github.workspace }}/.conan/conan_paths.cmake
-        --preset test
+        -DCMAKE_BUILD_TYPE=RelWithDebInfo
 
-    - name: Build, install and test
+        cmake --build build
+
+        cmake --install build
+
+    - name: Upload archive
+      uses: actions/upload-artifact@v3
+      with:
+        name: openassetio-${{ matrix.config.os }}
+        path: |
+          ${{ github.workspace }}/dist
+
+    # Reconfigure to add the test target. Ctest should rebuild using
+    # the cached build from prior.
+    - name: Test
       run: >
         ${{ matrix.config.preamble }}
 
-        ctest -VV --test-dir build --parallel 2
+        cmake --preset test build
+
+        ctest -VV --test-dir build --parallel 2 --build-config RelWithDebInfo
```

Comment:
[consider][minor] Missed a `--build-config` :grin:

Reply:
https://github.com/OpenAssetIO/OpenAssetIO/pull/1189/commits/16fe37d4ea80d06bf5175a2e32d26c631d36602d

Just gonna go ahead and merge since this was so minor.